fiscalyear.py

```diff
@@ -38,43 +34,52 @@ def fiscal_calendar(start_year=None,
     :raises ValueError: If ``start_month`` or ``start_day`` is out of range
     """
-    global START_YEAR
-    global START_MONTH
-    global START_DAY
-
-    # Use default values if not changed
-    if start_year is None:
-        start_year = START_YEAR
-    if start_month is None:
-        start_month = START_MONTH
-    if start_day is None:
-        start_day = START_DAY
-
-    assert isinstance(start_year, str)
-    assert start_year == 'previous' or start_year == 'same'
-    start_month = _check_month(start_month)
-    start_day = _check_day(start_month, start_day)
-
-    # Backup previous values
-    old_start_year = START_YEAR
-    old_start_month = START_MONTH
-    old_start_day = START_DAY
-
-    # Temporarily change global variables
+    if not isinstance(start_year, str):
+        raise TypeError("'start_year' must be a 'str', not: '%s'" % type(str))
+    if not start_year in ('previous',  'same'):
+        raise ValueError("start_year must be either 'previous' or 'same', not: '%s'" % start_year)
+    _check_day(start_month, start_day)
+
+
+def setup_fiscal_calendar(start_year, start_month, start_day):
+    validate_fiscal_calendar_params(start_year, start_month, start_day)
+    global START_YEAR, START_MONTH, START_DAY
     START_YEAR = start_year
     START_MONTH = start_month
     START_DAY = start_day
 
-    yield
 
-    # Restore previous values
-    START_YEAR = old_start_year
-    START_MONTH = old_start_month
-    START_DAY = old_start_day
+@contextlib.contextmanager
+def fiscal_calendar(start_year=None, start_month=None, start_day=None):
+    """A context manager that lets you modify the start of the fiscal calendar
+    inside the scope of a with-statement.
+
+    :param start_year: Relationship between the start of the fiscal year and
+        the calendar year. Possible values: ``'previous'`` or ``'same'``.
+    :type start_year: str
+    :param start_month: The first month of the fiscal year
+    :type start_month: int or str
+    :param start_day: The first day of the first month of the fiscal year
+    :type start_day: int or str
+    :raises ValueError: If ``start_year`` is not ``'previous'`` or ``'same'``
+    :raises TypeError: If ``start_month`` or ``start_day`` is not an int or int-like string
+    :raises ValueError: If ``start_month`` or ``start_day`` is out of range
+    """
+    # if arguments are omitted, use the currently active values.
+    start_year = START_YEAR if start_year is None else start_year
+    start_month = START_MONTH if start_month is None else start_month
+    start_day = START_DAY if start_day is None else start_day
+
+    # Temporarily change global variables and then restore previous values
+    old_start_year, old_start_month, old_start_day = START_YEAR, START_MONTH, START_DAY
+    setup_fiscal_calendar(start_year, start_month, start_day)
+    yield
+    setup_fiscal_calendar(old_start_year, old_start_month, old_start_day)
```

Comment:
I would keep the old comments and separate each step out line-by-line. I think it makes it more obvious what is going on, not that it's complicated.